Library/Homebrew/os/mac/sdk.rb

```diff
@@ -0,0 +1,57 @@
+require "os/mac/version"
+
+module OS
+  module Mac
+    class SDK
+      attr_reader :version, :path
+
+      def initialize version, path
+        @version = OS::Mac::Version.new version
+        @path = Pathname(path)
+      end
+    end
+
+    class SDKLocator
+      class NoSDKError < StandardError; end
+
+      def sdk_for v
+        cache_paths if @sdk_paths.nil?
+
+        path = @sdk_paths[v]
+        raise NoSDKError if path.nil?
+
+        SDK.new v, path
+      end
+
+      def latest_sdk
+        cache_paths if @sdk_paths.nil?
+        return if @sdk_paths.empty?
+
+        v, path = @sdk_paths.max
+        SDK.new v, path
+      end
+
+      private
+
+      def cache_paths
+        @sdk_paths = {}
+
+        # Xcode.prefix is pretty smart, so let's look inside to find the sdk
+        path = "#{Xcode.prefix}/Platforms/MacOSX.platform/Developer/SDKs"
+        # Xcode < 4.3 style
+        path = "/Developer/SDKs" if !File.directory? path
+        # Finally query Xcode itself (this is slow, so check it last)
+        path = File.join(Utils.popen_read(OS::Mac.locate("xcrun"), "--show-sdk-platform-path").chomp, "Developer", "SDKs") if !File.directory? path
```

Comment:
This feels too cramped and the trailing `if` modifier could be easily overlooked. Maybe convert this to an `if` block and also split the `File.join` and `Utils.popen_read` invocations into separate lines?